charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml

```diff
@@ -274,6 +277,29 @@ data:
       - target_label: instance
         replacement: kube-state-metrics
       metric_relabel_configs:
+      # We want to drop all metrics that
+      # do not belong to the shoot
+
+      # Add a temp label to all ksm metrics
+      - target_label: __tmp_keep
+        replacement: false
+      # Keep all shoot metrics
+      - source_labels: [type]
+        regex: shoot
+        target_label: __tmp_keep
+        replacement: true
+      # we want to keep all metrics belonging to the shoot
+      - source_labels: [namespace]
```

Comment:
Hmm, the Shoot ksm also collects metrics for `nodes`:
https://github.com/gardener/gardener/blob/3b4ae6caf2fb60731d5481e63b2d8fe577760b90/charts/seed-bootstrap/charts/kube-state-metrics/templates/deployment.yaml#L40
This will be filtered out now because they probably don't have a namespace, right?
If we need those metrics, would it make sense to only collect them once in the seed ksm?

Reply:
That is correct but we still need those because the aggregate prometheus collects these for seed monitoring (indirectly via the `prometheus-0` in the garden ns).

Reply:
Sorry, I got this mixed up. I looked at the seed ksm, not the shoot ksm 🙈
So this is all fine... :)